src/factory-client.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const request = require('superagent')
+const DaemonClient = require('./ipfsd-client')
+
+/*
+ * Exposes the same Factory API but uses a remote endpoint to create the Daemons/Nodes
+ */
+class FactoryClient {
+  constructor (opts) {
+    opts = opts || {}
+    if (!opts.host) {
+      opts.host = 'localhost'
+    }
+
+    if (!opts.port) {
+      opts.port = 9999
+    }
+
+    if (typeof opts.host === 'number') {
+      opts.port = opts.host
+      opts.host = 'localhost'
+    }
+
+    this.port = opts.port
+    this.host = opts.host
+    this.type = opts.type || 'go'
+
+    if (this.type === 'proc') {
+      throw new Error(`'proc' is not allowed in remote mode`)
+    }
+
+    this.baseUrl = `${opts.secure ? 'https://' : 'http://'}${this.host}:${this.port}`
+  }
+
+  /**
+   * Get the version of the IPFS Daemon.
+   *
+   * @memberof FactoryDaemon
+   * @param {Object} [options={}]
+   * @param {function(Error, string)} callback
+   * @returns {undefined}
+   */
+  version (options, callback) {
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+    // TODO implement this method
```

Comment:
- [ ] This method needs to be implemented as well for clients. Once it is, enable tests again.

Reply:
In progress, will push shortly.